prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -4726,7 +4726,7 @@ subroutine checkReksConsistency(reksIni, onSiteElements, kPoint, nEl, nKPoint, t
       end if
     end if
 
-    if (reksIni%Efunction /= 1 .and. tLatOpt) then
+    if (reksInp%Efunction /= 1 .and. tLatOpt) then
```

Comment:
Is there a reason not to also use an enumerated type for the energy function?

Reply:
In `Energy` block of current input, it reads string list such as {"PPS" "OSS"}. This options can be different when we use SSR(4,4) case, which means I think that we have to increase enumerated types when the active space size increases.

For example, SSR(4,4) can use the following functionals.
1) only PPS (which is single-state REKS in (4,4) case)
2) PPS + OSS1 + OSS2,  3) PPS + DSPS,  4) and so on.

Thus, the second option in SSR(2,2) is PPS + OSS, but the second one in SSR(4,4) becomes PPS + OSS1 + OSS2. So, I do not use enumerated type for `reksInp%Efunction` at the moment.

Reply:
@bhourahine How about this? I think that we can sort the enumerated type like this. The ten-th digit indicates the active space size (kind) and the digit indicates the possible choice. For example, certain enumerated type `reksEfunc` = 0 for PPS and = 1 for PPS + OSS in SSR(2,2) case. Likewise, `reksEfunc` = 10 for PPS and 11 for PPS + OSS1 + OSS2 and 12 for PPS + DSPS and so on. If you want to agree this opinion, I will try to change the `Efunction` to `reksEfunc` enumerated type.

Reply:
Seems reasonable, but assumptions about order or additivity of the constants outside of the dedicated reks routines should be removed.

Reply:
Then I will maintain current status for this parser routine. Is there other comments? If not I will change the documentation from now on.